elyra/pipeline/component_registry.py

```diff
@@ -223,8 +259,15 @@ def get_component(self, component_id: str) -> Component:
         cached_component = next((component for component in self._cache if component.id == component_id), None)
```

Comment:
Looks like this returns `None` if there isn't a component_id match.  If so, please update the return type to `Optional[Component]`